blockchain/chain.go

```diff
@@ -1683,6 +1690,26 @@ func (b *BlockChain) MainChainHasBlock(hash *chainhash.Hash) bool {
 	return node != nil && b.bestChain.Contains(node)
 }
 
+// Ancestor returns whether the provided hash is the ancestor of the provided
+// block at the provided height.
+//
+// This function is safe for concurrent access.
+func (b *BlockChain) Ancestor(block *chainhash.Hash, height int64, hash *chainhash.Hash) bool {
+	node := b.index.LookupNode(block)
+	ancestor := node.Ancestor(height)
+	return ancestor.hash.IsEqual(hash)
+}
+
+// RemoveSpendEntry purges the associated spend journal entry to the
+// provided hash.
+//
+// This function is safe for concurrent access.
+func (b *BlockChain) RemoveSpendEntry(hash *chainhash.Hash) error {
```

Comment:
I agree that it's misleading and potentially dangerous to have this on the public `BlockChain` API since it shouldn't just be used by any caller.  My original suggestion was to have `SpendJournalEntry` accept this as a function type on creation, in which case it would not need to be exposed like this.

Reply:
I think the idea of having `internalChainAdapter` is better overall for this because the pruner will still need access to other blockchain functions. Would be better to do `NewSpendJournalPruner(chainAdapter)` than `NewSpendJournalPruner(chainAdapter, removeSpendEntry)` considering `removeSpendEntry` could be part of the adapter.